### ISSUE TITLE
Incorrect paths during configure of core extensions

### ISSUE THREAD

**Issue by csandanov** [docker-library/php#549]
I forked `_/php` and got an error during compilation of phar extension. I found out that the problem was in incorrect paths in the extension's Makefile.  Usually, this problem doesn't appear because source's and target's timestamps match and this step omit, in my case they didn't (randomness of tar extract?). I submitted a [PR](https://github.com/php/php-src/pull/2955) to `php/php-src` and it seems that the real problem is that:

> The extensions in the core are actually not foreseen for phpize.

More details in [this comment](https://github.com/php/php-src/pull/2955#issuecomment-352814935)

**Comment by tianon** [comment]
Indeed, there have been a few similar issues like this, although I don't recall seeing one that's quite so directly tied to _paths_ like this one is (usually it's oddities about how the `./configure` script functions during PHP's build process vs `phpize`'s).

Regardless, I've left a comment on the PR to hopefully help the kind folks from upstream understand _why_ we do these crazy things. :sweat_smile:

**Comment by tianon** [comment]
Definitely seems like you're onto something a timestamp ghost though -- I can't seem to reproduce the failure.  I installed `libssl-dev`, then `docker-php-ext-install phar` works every time. :disappointed:

Additionally though, `phar` is already built-in, from what I can tell:
```console
$ docker run -it --rm php:7.2-cli bash
root@82029bff1720:/# php -i | grep -i phar
Registered PHP Streams => https, ftps, compress.zlib, php, file, glob, data, http, ftp, phar
Phar
Phar: PHP Archive support => enabled
Phar EXT version => 2.0.2
Phar API version => 1.1.1
Phar-based phar archives => enabled
Tar-based phar archives => enabled
ZIP-based phar archives => enabled
Phar based on pear/PHP_Archive, original concept by Davey Shafik.
Phar fully realized by Gregory Beaver and Marcus Boerger.
phar.cache_list => no value => no value
phar.readonly => On => On
phar.require_hash => On => On
```
